lib/model/model.go

```diff
@@ -2445,29 +2445,32 @@ func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly
 			return true
 		}
 
-		fullName := strings.Split(f.Name, sep)
-		name := fullName[len(fullName)-1]
-		path := fullName[:len(fullName)-1]
+		name := filepath.Base(f.Name)
+		path := filepath.Dir(f.Name)
 
-		tree := DirectoryTree{Name: name, IsDirectory: f.IsDirectory(), Children: []*DirectoryTree{}}
-		for !reflect.DeepEqual(path, currentPrefix) && len(currentPrefix) != 0 {
+		var children []*FolderItem = nil
+		if f.FileType() == 1 {
+			children = make([]*FolderItem, 0)
+		}
+		tree := FolderItem{Name: name, FileType: f.FileType(), Children: children}
```

Comment:
Could do f.FileType().String().

Also, as I suggested, children can always be nil (simply omitted from declaration all together), due to how append works..